src/classes/ALLO_Allocations_TDTM.cls

```diff
@@ -226,18 +225,18 @@ public class ALLO_Allocations_TDTM extends TDTM_Runnable {
 
             //INSERT case
             if (triggerAction == TDTM_Runnable.Action.AfterInsert) {
-                //Don't do anything on insert if we don't have an amount or if amount is $0
-                if ((pmt.npe01__Payment_Amount__c != null) && (pmt.npe01__Payment_Amount__c != 0)) {
+                //Don't do anything on insert if we don't have an amount
+                if ((pmt.npe01__Payment_Amount__c != null)) {
                     pmtsNeedingAllocations.add(pmt);
                 }
             }
 
             //UPDATE case: if the payment amount changed, add to the list for processing
             if (triggerAction == TDTM_Runnable.Action.AfterUpdate) {
-                Decimal oldAmount = valueOrDefaultIfNull((Decimal) oldlist[i].get('npe01__Payment_Amount__c'), 0);
-                Decimal newAmount = valueOrDefaultIfNull(pmt.npe01__Payment_Amount__c, 0);
+                Decimal oldAmount = (Decimal) oldlist[i].get('npe01__Payment_Amount__c');
```

Comment:
I think this will throw an NPE if `npe01__Payment_Amount__c` is null. Why not just use the following this way it doesn't crash if the value is null
`Decimal oldAmount = oldlist[i].npe01__Payment_Amount__c;`

Reply:
Thanks for the feedback.  I am pretty sure the get will just return null if there is no value present in the field map.  I can change to your suggested format since it is easier to read, but I believe I just mirrored what was already done in the Opportunity part of the triggers (I will double check though).

Reply:
Thanks. It's true that `oldlist[i].npe01__Payment_Amount__c` will return null, however casting the null as a Decimal (`(Decimal)null`) is what will throw the exception. Since the field always exists on the record, it's more efficient to just retrieve it directly and avoid the casting.

Reply:
I am happy to make the change you are asking me to make, but as a point of order what you are stating is incorrect.  You absolutely CAN cast null to a Decimal type, or virtually any other.  It will not throw an exception.  The value will simply be null. 

Whether you should or not is a different question...

Reply:
I stand very corrected. I could have sworn that this used to throw an exception - specifically when an object type is cast to something concrete. It definitely works fine now, so no need to change anything here.

Reply:
Hehe.  You are awesome!!!  No worries on my end.  I so value your feedback, and really appreciate the chance to talk back and forth when there is something like this!!!!